src/entities/nestedJoin/types.ts

```diff
@@ -0,0 +1,48 @@
+import { Address, Hex } from '../../types';
+import { Slippage } from '../slippage';
+import { Token } from '../token';
+import { TokenAmount } from '../tokenAmount';
+
+export type NestedJoinInput = {
+    amountsIn: {
+        address: Address;
+        rawAmount: bigint;
+    }[];
+    chainId: number;
+    rpcUrl: string;
+    accountAddress: Address;
+    useNativeAssetAsWrappedAmountIn?: boolean;
+    fromInternalBalance?: boolean;
+};
+
+export type NestedJoinCallAttributes = {
+    chainId: number;
+    useNativeAssetAsWrappedAmountIn: boolean;
+    sortedTokens: Token[];
+    poolId: Hex;
+    poolAddress: Address;
+    poolType: string;
+    kind: number;
```

Comment:
Could maybe make an enum type here to make it easier to follow what the different kinds are?